### PR TITLE
Signing artifacts

### PR DESCRIPTION
**Pull request by AbhijithGanesh**
## Description

Fixes pyrsia/pyrsia#511

This PR does provides an opportunity to sign artifacts by signing it with libp2p ed25519 keys and it 

## PR Checklist

- [ ] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [ ] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [ ] I've included a good title and brief description along with how to review them.
- [ ] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [ ] I've requested a review  from `pyrsia/collaborators`.

### Code Contributions

- [ ] I've built the code `cargo build --all-targets` successfully.
- [ ] I've run the unit tests `cargo test --workspace` and everything passes.
- [ ] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Comment by chb0github** [comment]
I don't get the point of this code. It's 2 functions identical to the underlying code, which is implemented as a single line. So, help me understand purpose

**Comment by AbhijithGanesh** [comment]
Currently, the functions don't take any default parameter, I was under the assumption public key will be shared and it will be passed as default value. The value will then be used to verify whenever the function is being called.

File: src/signed.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (MitaliBo) [inline src/signed.rs:1]: To adhere with the project structure we want to move any any new modules in a directory and then create a mod file for that.

File: src/signed.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+*/
+use libp2p::core::identity::ed25519::{Keypair, PublicKey};
+
+pub fn sign(key: &Keypair, payload: &[u8]) -> Vec<u8> {
+    return key.sign(payload);
+}
+
+pub fn verify(key: &PublicKey, message: &[u8], signature: Vec<u8>) -> bool {
+    return key.verify(message, &signature);
+}
```
Comment (MitaliBo) [inline src/signed.rs:24]: why do we want these wrapper functions ?

**Review by MitaliBo — CHANGES_REQUESTED** [review]

**Comment by betarelease** [comment]
At this time we are still figuring out what signing means for Pyrsia (notes here - https://docs.google.com/document/d/1mOSMyZkM-HaKYnKNwk_Qr6t3331EZXHSQgp2L8dsjw4/edit#)

We plan to put this on hold for now and take this up again when we have a need for signing.

**Comment by betarelease** [comment]
Closing this pull request after discussion during the sprint planning and has been stale for awhile.

**Pull request closed without merging**